File: backend/app/schemas/expert.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel


# ── REST ─────────────────────────────────────────────────

class ExpertQueueItem(BaseModel):
    id: str
    session_id: str
    user_id: str
    risk_level: int
    risk_factors: list[str]
    ai_response: str
    context_summary: Optional[str]
    status: str  # "pending" | "approved" | "modified"
    created_at: datetime

    class Config:
        from_attributes = True


class ExpertQueueResponse(BaseModel):
    items: list[ExpertQueueItem]
    total: int


class FeedbackRequest(BaseModel):
    session_id: str
    pending_response_id: str
    action: str                          # "approve" | "modify"
    modified_content: Optional[str] = None
    feedback_category: Optional[str] = None  # "response_quality"|"safety"|"clinical_accuracy"
    feedback_note: Optional[str] = None


class FeedbackResponse(BaseModel):
    id: str
    status: str
    reviewed_at: datetime


# ── WebSocket ─────────────────────────────────────────────

class WSHighRiskAlert(BaseModel):
    type: str = "high_risk_alert"
    review_id: str
    session_id: str
    user_id: str
    risk_level: int
    risk_factors: list[str]
    ai_response: str
    context_summary: Optional[str] = None


class WSReviewDecision(BaseModel):
    type: str  # "review_decision"
    pending_response_id: str
    action: str             # "approve" | "modify"
    modified_content: Optional[str] = None
    feedback_note: Optional[str] = None

File: frontend/src/api/expert.ts
import { api } from './client'

export interface ReviewItem {
  id: string
  session_id: string
  user_id: string
  risk_level: number
  risk_factors: string[]
  ai_response: string
  context_summary: string | null
  status: 'pending' | 'approved' | 'modified'
  created_at: string
}

export interface FeedbackPayload {
  session_id: string
  pending_response_id: string
  action: 'approve' | 'modify'
  modified_content?: string
  feedback_category?: 'response_quality' | 'safety' | 'clinical_accuracy'
  feedback_note?: string
}

export const expertApi = {
  getQueue: (token: string) =>
    api.get<{ items: ReviewItem[]; total: number }>(`/expert/queue?token=${token}`),

  submitFeedback: (payload: FeedbackPayload, token: string) =>
    api.post(`/expert/feedback?token=${token}`, payload),
}
